src/software/ai/ai.h

```diff
@@ -48,24 +48,31 @@ class AI final
      */
     TbotsProto::PlayInfo getPlayInfo() const;
 
-   private:
     /**
-     * Overrides the play from the name
+     * Overrides the play from the play proto
      *
-     * @param name the play name
-     * @param ai_config
+     * @param play_proto the play proto
      */
-    void overridePlayFromName(std::string name);
+    void overridePlayFromProto(TbotsProto::Play play_proto);
 
+    /**
+     * Update the AiConfig proto
+     *
+     * @param ai_config The new AiConfig proto
+     */
+    void updateAiConfig(TbotsProto::AiConfig ai_config);
+
+   private:
     void checkAiConfig();
+    bool ai_config_changed = false;
```

Comment:
this should be initialized in the constructor

Reply:
done